WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java

```diff
@@ -177,12 +178,16 @@ public static void addMedia(Activity activity) {
     }
 
     public static void viewMyProfile(Context context) {
+        AccountHelper.getDefaultAccount().fetchAccountSettings();
+
         Intent intent = new Intent(context, MyProfileActivity.class);
         AnalyticsUtils.trackWithCurrentBlogDetails(AnalyticsTracker.Stat.OPENED_MY_PROFILE);
         slideInFromRight(context, intent);
     }
 
     public static void viewAccountSettings(Activity activity) {
+        AccountHelper.getDefaultAccount().fetchAccountSettings();
```

Comment:
I would actually prefer my first suggestion, which is to be closer to the caller.

Here's the thing: `viewAccountSettings` seems innocent enough (it has "view" in its name) and is in an activity launcher helper class. But, it also changes the local world since it fetches and updates local data. That's quite different than spawning an analytics event and it couples the activity opening with a network action. I would feel way more comfortable if those two operations were not coupled like that :)